src/configurator/x86_64/pvh.rs

```diff
@@ -164,6 +173,16 @@ impl BootConfigurator for PvhBootConfigurator {
             .write_slice(params.header.as_slice(), params.header_start)
             .map_err(|_| Error::StartInfoSetup)?;
 
+        if let Some(modules) = params.modules.as_ref() {
+            let modules_addr = params.modules_start.ok_or(Error::ModulesAddressMissing)?;
+            guest_memory
+                .checked_offset(modules_addr, modules.len())
+                .ok_or(Error::ModulesPastRamEnd)?;
+            guest_memory
+                .write_slice(modules.as_slice(), modules_addr)
+                .map_err(|_| Error::ModulesSetup)?;
```

Comment:
Mh, the first check technically only verifies that `modules_addr + modules.len()` is a valid guest address, it doesn't check anything about range `[modules_addr, module_addr + modules.len())`. However, `write_slice` already does verify the entire range, so if we do not particularly care about having a separate error variant, we can just drop the `checked_offset` check. Otherwise, we can do 
 ```suggestion
            let slice = guest_memory
                 .get_slice(modules_addr, modules.len())
                 .ok_or(Error::ModulesPastRamEnd)?;
             slice
                 .write_slice(modules.as_slice(), modules_addr)
                 .map_err(|_| Error::ModulesSetup)?;
```

Reply:
Good point! Removed the extra check.